### PR TITLE
feat: Add role prop to Text component

### PR DESCRIPTION
**Pull request by gabrieldonadel**
## Summary

As pointed out by @necolas on https://github.com/facebook/react-native/issues/34424#issuecomment-1261482517 we forgot we add the `role` prop mapping to the `Text` component. This PR adds a new `role` prop to `Text`, mapping the web `role` values to the already existing `accessibilityRole` prop and moves the `roleToAccessibilityRoleMapping` to a common file that can be imported by both the `Text` and `View` components as requested on https://github.com/facebook/react-native/issues/34424. This PR also updates the RNTester AcessebilityExample to include a test using this new prop.

## Changelog

[General] [Added] - Add role prop to Text component

## Test Plan
 
1. Open the RNTester app and navigate to the Accessibility Example page
2. Test the `role` prop through the `Text with role = heading` section

### CODE REVIEW

File: Libraries/Components/View/ViewAccessibility.d.ts [hunk 1]
```diff
@@ -286,3 +291,69 @@ export interface AccessibilityPropsIOS {
    */
   accessibilityIgnoresInvertColors?: boolean | undefined;
 }
+
+export type Role =
```
Comment (gabrieldonadel) [inline Libraries/Components/View/ViewAccessibility.d.ts:295]: These were missing on TS

File: Libraries/Utilities/AcessibilityMapping.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow strict-local
+ * @format
+ */
+
+'use strict';
+
+// Map role values to AccessibilityRole values
+export const roleToAccessibilityRoleMapping = {
```
Comment (jacdebug) [inline Libraries/Utilities/AcessibilityMapping.js:14]: This object `roleToAccessibilityRoleMapping` needs to in JS VM heap all time after loading this JS. Consider  making this is getter function so this object can be deallocated after the call, or better use a switch statement like this. 
https://github.com/facebook/react-native/blob/84225573c17b17844b3b3e2a438b3178f7d35b05/packages/normalize-color/index.js#L303

cc @javache

Comment (gabrieldonadel) [reply Libraries/Utilities/AcessibilityMapping.js]: Sure, I've just refactor it to be a switch statement

**Comment by gabrieldonadel** [comment]
@jacdebug would you mind rereviwing this PR?

**Review by jacdebug — APPROVED** [review]
Thanks for the fixups, LGTM and sorry for delay!

**Review by necolas — APPROVED** [review]

**Pull request closed without merging**